nautobot/core/api/parsers.py

```diff
@@ -0,0 +1,133 @@
+import csv
+from io import StringIO
+import json
+import logging
+
+from django.conf import settings
+
+from rest_framework import serializers
+from rest_framework.exceptions import ParseError
+from rest_framework.parsers import BaseParser
+
+from nautobot.core.models.utils import deconstruct_natural_key_slug
+
+
+logger = logging.getLogger(__name__)
+
+
+class NautobotCSVParser(BaseParser):
+    """Counterpart to NautobotCSVRenderer - import CSV data."""
+
+    media_type = "text/csv"
+
+    def parse(self, stream, media_type=None, parser_context=None):
+        parser_context = parser_context or {}
+        encoding = parser_context.get("encoding", "UTF-8")
+        try:
+            if "serializer_class" in parser_context:
+                # UI bulk-import case
+                serializer_class = parser_context["serializer_class"]
+            else:
+                # REST API case
+                serializer_class = parser_context["view"].get_serializer_class()
+        except (KeyError, AttributeError):
+            logger.error("Unable to find serializer_class for the view?")
```

Comment:
Probably need to rework this to provide more context to the error, if it's even necessary. I don't think I've actually hit this error or the one below for any of our views/serializers since I initially got this to work. It **might** be better to just raise a `ParseError` instead of returning None. I'll do some investigation.

Reply:
Changing to a ParseError.